### PR TITLE
[IOPID-917] : make exchange token valid to see profile

### PR DESCRIPTION
**Pull request by pelliccm**
#### List of Changes

Let the user see their profile even if they logged in with an exchange token, via magic link

#### Motivation and Context

Request to see the profile when logged in via magic link

#### How Has This Been Tested?

Unit test, Postman

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ClaudioNTT** [comment]
## Adjustment according to the table

image

File: Profile/__test__/handler.test.ts [hunk 1]
```diff
@@ -27,6 +29,12 @@ const aValidUser: HslJwtPayloadExtended = {
   name: "name" as NonEmptyString,
   spid_level: SpidLevel.L2
 };
+const aValidExchangeUser: ExchangeJwtPayloadExtended = {
```
Comment (arcogabbo) [inline Profile/__test__/handler.test.ts]: may I suggest to put this in `config.mock.ts` and importing it among all test files that needs it? in this PR i saw this exact object declared in 3 different test files

File: SessionState/handler.ts [hunk 2]
```diff
@@ -1,66 +1,77 @@
...
 export const sessionStateHandler = (
   client: SessionStateClient
 ): SessionStateHandlerT => (
-  reqJwtPayload: HslJwtPayloadExtended
+  reqJwtPayload: HslJwtPayloadExtended | ExchangeJwtPayloadExtended
 ): ReturnType<SessionStateHandlerT> =>
   pipe(
     reqJwtPayload,
     TE.fromPredicate(
       user_data => canSeeProfile(user_data),
```
Comment (BurnedMarshal) [inline SessionState/handler.ts]: We can use a more compact style
```suggestion
      canSeeProfile,
```

**Review by BurnedMarshal — APPROVED** [review]
LGTM apart a minor change

**Pull request merged**